packages/neon_framework/packages/account_repository/test/account_repository_test.dart

```diff
@@ -0,0 +1,602 @@
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList.first)),
+          ),
+        );
+
+        verifyNever(() => storage.readLastAccount());
+      });
+
+      test('uses last active account when specified', () async {
+        when(() => storage.readCredentials()).thenAnswer((_) async => credentialsList);
+        when(() => storage.readLastAccount()).thenAnswer((_) async => credentialsList[1].id);
+        await repository.loadAccounts(rememberLastUsedAccount: true);
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList[1])),
+          ),
+        );
+
+        verify(() => storage.readLastAccount()).called(1);
+      });
+
+      test('ignores last active account when not in accounts', () async {
+        when(() => storage.readCredentials()).thenAnswer((_) async => credentialsList);
+        when(() => storage.readLastAccount()).thenAnswer((_) async => 'invalid');
+        await repository.loadAccounts(rememberLastUsedAccount: true);
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList.first)),
+          ),
+        );
+
+        verify(() => storage.readLastAccount()).called(1);
+      });
+
+      test('prefers last active account over initial account', () async {
+        when(() => storage.readCredentials()).thenAnswer((_) async => credentialsList);
+        when(() => storage.readLastAccount()).thenAnswer((_) async => credentialsList[1].id);
+        await repository.loadAccounts(
+          rememberLastUsedAccount: true,
+          initialAccount: credentialsList.first.id,
+        );
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', containsAllInOrder(accountsList))
+                .having((e) => e.active, 'active', equals(accountsList[1])),
+          ),
+        );
+
+        verify(() => storage.readLastAccount()).called(1);
+      });
+    });
+
+    group('getServerStatus', () {
+      test('fetches the server status', () async {
+        final mockResponse = _DynamiteResponseMock<core.Status, void>();
+        when(() => coreClient.getStatus()).thenAnswer((_) async => mockResponse);
+        when(() => mockResponse.body).thenReturn(_FakeStatus());
+
+        expect(
+          repository.getServerStatus(Uri()),
+          completion(isA<core.Status>()),
+        );
```

Comment:
For example

Reply:
You are right.
This wasn't an issue because there is no method call we try to validate after the async suspension.

I'll change them nevertheless to allow future alterations.